src/pages/index.js

```diff
@@ -24,10 +46,123 @@ const profileEditButton = document.querySelector("#profile-edit-button");
 const profileCardModal = document.querySelector("#profile-card-modal");
 const profileEditModal = document.querySelector("#profile-edit-modal");
 const profileEditForm = profileEditModal.querySelector("#editProfileForm");
+const saveButton = document.querySelector(".modal__save-button");
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__picture"
+);
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData.name, userData.about, userData.avatar);
+  })
+  .catch((err) => {
+    console.error("Error fetching user info:", err);
+  });
+
+profileEditButton.addEventListener("click", () => {
+  const currentUserInfo = userInfo.getUserInfo();
+  profilePopup.setInputsValues(currentUserInfo);
+  profilePopup.open();
+});
+
+const profilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileFormSubmit
+);
+profilePopup.setEventListeners();
+
+function handleProfileFormSubmit(formValues) {
+  showSavingText(saveButton);
+  console.log("Submitting form with values:", formValues); // Log the form values
+  api
+    .editUserInfo(formValues)
+    .then((editUserInfo) => {
+      console.log("Profile updated:", editUserInfo); // Log the updated user info
+      userInfo.setUserInfo(
+        editUserInfo.name,
+        editUserInfo.about,
+        editUserInfo.avatar
+      );
+
+      profilePopup.close();
+      hideSavingText(saveButton);
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    });
+}
+
+/*************************
+ * Profile Picture Modal *
+ *************************/
+
+const profilePictureButton = document.querySelector(".pencil-icon");
+
+profilePictureButton.addEventListener("click", () => {
+  profilePicturePopup.open();
+});
+
+const profilePicturePopup = new PopupWithForm(
+  "#profile-picture-modal",
+  handleProfilePictureFormSubmit
+);
+profilePicturePopup.setEventListeners();
+// Selectors for profile picture form
+const profilePictureForm = document.querySelector(
+  "#profile-picture-modal form"
+);
+const profilePictureInput = profilePictureForm.querySelector(
+  'input[name="newProfilePictureUrl"]'
+);
+const profilePictureSaveButton = profilePictureForm.querySelector(
+  'button[type="submit"]'
+);
+
+// Disable save button if form is empty
+profilePictureInput.addEventListener("input", () => {
```

Comment:
Please, delete it

Now only class `FormValidator` should **`disable/enable`** submit buttons and remove errors from the inputs (if you do it)